indico/modules/events/requests/__init__.py

```diff
@@ -49,7 +50,10 @@ def _merge_users(target, source, **kwargs):
 @signals.event.deleted.connect
 def _event_deleted(event, **kwargs):
     from indico.modules.events.requests.models.requests import Request, RequestState
+
+    is_past_event = event.start_dt < now_utc()
+
     query = (Request.query.with_parent(event)
              .filter(Request.state.in_((RequestState.accepted, RequestState.pending))))
     for req in query:
-        req.definition.withdraw(req, notify_event_managers=False)
+        req.definition.withdraw(req, notify_event_managers=False, notify_request_managers=not is_past_event)
```

Comment:
Just for better readability:

```suggestion
        req.definition.withdraw(req, notify_event_managers=False, notify_request_managers=(not is_past_event))
```